catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -131,22 +131,31 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
             Map<String, Object> headers) throws SourceUnavailableException, IngestException {
         LOGGER.debug("Creating content item.");
 
-        String key = String.valueOf(ingestedFile.getAbsolutePath()
-                .hashCode());
-        String id = fileIdMap.loadAllKeys()
-                .contains(key) ? String.valueOf(fileIdMap.loadFromPersistence(key)) : null;
-
+        String refKey = (String) headers.get(Constants.STORE_REFERENCE_KEY);
+        String safeKey = null;
+        String id = null;
+        String name = ingestedFile.getName();
+
+        if (refKey != null) {
+            name = FilenameUtils.getName(refKey);
+            // guards against impermissible filesystem characters
+            safeKey = String.valueOf(refKey.hashCode());
+            if (fileIdMap.loadAllKeys()
+                    .contains(safeKey)) {
+                id = String.valueOf(fileIdMap.loadFromPersistence(safeKey));
+            }
+        }
         if (StandardWatchEventKinds.ENTRY_CREATE.equals(eventType)) {
             CreateStorageRequest createRequest =
                     new CreateStorageRequestImpl(Collections.singletonList(new ContentItemImpl(
                             uuidGenerator.generateUuid(),
                             Files.asByteSource(ingestedFile),
                             mimeType,
-                            ingestedFile.getName(),
+                            name,
                             0L,
                             null)), null);
```

Comment:
The `processHeaders` sets properties on the RequestImpl objects. Change line 156 from:

`null)), null);`

to:

` null)), Maps.transformValues(headers, (v) -> (Serializable) v));`

The properties are set. Let Guava do the work of changing `Map<String, Object>` to `Map<String, Serializable>`. Then you can remove the `processHeaders` method.